pkg/utils/utils_test.go

```diff
@@ -396,35 +403,89 @@ var _ = Describe("In the utils package", func() {
 	)
 
 	DescribeTable("getting PF names",
-		func(fs *FakeFilesystem, device string, expected string, shouldFail bool) {
+		func(fs *FakeFilesystem, eswitchMode string, device string, expected string, shouldFail bool) {
+			fakeNetlinkProvider := mocks.NetlinkProvider{}
+			fakeNetlinkProvider.
+				On("GetDevLinkDeviceEswitchAttrs",
```

Comment:
can you add also a case for when devlink device attributes return eswitch mode `legacy` ?

you can split the tables to "getting PF names legacy", "getting PF names switchdev" if you feel its getting to complex with the number of args here.

also maybe use other device name to trigger error like "nonDevlinkDevice", "devlinkDeviceSwitchdev", "devlinkDeviceLegacy" ?

passing device name as "eswitchUnsupportedError"/"unknownError" is a bit unclear to read imo.

Reply:
Done